core/src/components/router/utils/path.ts

```diff
@@ -27,33 +48,20 @@ export const chainToPath = (chain: RouteChain): string[] | null => {
   return path;
 };
 
-export const writePath = (history: History, root: string, useHash: boolean, path: string[], direction: RouterDirection, state: number, queryString?: string) => {
-  let url = generatePath([
-    ...parsePath(root).segments,
-    ...path
-  ]);
-  if (useHash) {
-    url = '#' + url;
-  }
-  if (queryString !== undefined) {
-    url = url + '?' + queryString;
-  }
-  if (direction === ROUTER_INTENT_FORWARD) {
-    history.pushState(state, '', url);
-  } else {
-    history.replaceState(state, '', url);
-  }
-};
-
-export const removePrefix = (prefix: string[], path: string[]): string[] | null => {
+// Remove the prefix segments from the path segments.
+//
+// Return:
+// - null when the path segments do not start with the passed prefix,
+// - the path segments after the prefix otherwise.
+const removePrefix = (prefix: string[], path: string[]): string[] | null => {
   if (prefix.length > path.length) {
     return null;
   }
   if (prefix.length <= 1 && prefix[0] === '') {
     return path;
   }
   for (let i = 0; i < prefix.length; i++) {
-    if (prefix[i].length > 0 && prefix[i] !== path[i]) {
+    if (prefix[i] !== path[i]) {
```

Comment:
`length == 0` could only happen for when `prefix = ['']` - other empty parts are filtered out in `parsePath()`. This case is already accounted for on l. 60.

_I also removed the `export` to make sure this function is only called internally by the router_

Reply:
That sounds good. While we are changing this function, could we add a few tests?

Reply:
That a common debate: should we test "private" stuff (in langs that support private). The general consensus is no.

The function is kind of private (I removed the `export`) and is call only from `readPath` that is extensively tested.

If we want to test `removePrefix` then we would have to `export` it, sending the signal that it could be used anywhere and then it would be good to keep the `> 0` because we can not make any assumptions on what the arguments would be.

So I would say that:
1) it's not worth adding the `export`
2) this function is tested via `readPath`
3) I would even inline this in `readPath` but I don't know your policy about it.

Reply:
We can keep it as is. I am fine having it get tested via `readPath`.